src/benchmarks/micro/libraries/System.Diagnostics.Tracing/Perf_EventListener.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System.Runtime.CompilerServices;
+
+namespace System.Diagnostics.Tracing
+{
+    [BenchmarkCategory(Categories.Libraries)]
+    public class Perf_EventListener
+    {
+        private enum TestEnum
+        {
+            Foo = 123
+        }
+
+        private sealed class BenchmarkEventSource : EventSource
+        {
+            public BenchmarkEventSource() : base(nameof(Perf_EventListener)) { }
+
+            [Event(1)]
+            public void EventNoParams() => WriteEvent(1);
+
+            [Event(2)]
+            public void EventIntParams(int arg1, int arg2, int arg3) => WriteEvent(2, arg1, arg2, arg3);
+
+            [Event(3)]
+            public void EventStringParams(string arg1, string arg2, string arg3) => WriteEvent(3, arg1, arg2, arg3);
+
+            [Event(4)]
+#if NET
+            [SkipLocalsInit]
```

Comment:
>SkipLocalsInit [](http://example.com/codeflow?start=13&length=14)

Why you need to use it in the first place? You need to care about any regression moving forward. That means initializing the locals is const and shouldn't contribute to any future regression.

Reply:
Agreed. I don't see any problem with it but it wasn't clear why it was needed.

Reply:
It's not needed, it's just a better reflection of how it will behave in runtime code.
I agree it is not necessary in order to catch regressions.